infra/base-images/base-builder/cargo

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash -eu
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# we need this until https://github.com/rust-lang/cargo/issues/5450 or similar is merged
+# because cargo uses relative paths for the current crate
+# and absolute paths for its dependencies
+
+export PATH=/rust/bin:/usr/bin
+if [ "$SANITIZER" = "coverage" ] && [ $1 = "build" ]
+then
+    abspath=`cargo metadata --no-deps --format-version 1 | jq -r '.workspace_root'`
```

Comment:
maybe better var name than abspath, what is this abspath for?

Reply:
This is the absolute path of the source code of the crate getting built.

Reply:
maybe crate_src_abspath?

Reply:
ok good